src/GraphQL/Types/Composite/AutoRegisteringHelper.cs

```diff
@@ -224,5 +224,143 @@ internal static LambdaExpression GetParameterExpression(Type parameterType, Quer
+                    expressions.Add(expression);
+                }
+                var memberInstanceExpression = BuildMemberInstanceExpression(methodInfo);
+                if (IsObservable(methodInfo.ReturnType))
+                {
+                    var resolver = new SourceStreamMethodResolver(methodInfo, memberInstanceExpression, expressions);
+                    fieldType.Resolver = resolver;
+                    fieldType.StreamResolver = resolver;
+                }
+                else
+                {
+                    var resolver = new MemberResolver(methodInfo, memberInstanceExpression, expressions);
+                    fieldType.Resolver = resolver;
+                    fieldType.StreamResolver = null;
+                }
+                fieldType.Arguments = queryArguments;
+            }
+            else if (memberInfo is FieldInfo fieldInfo)
+            {
+                var resolver = new MemberResolver(fieldInfo, BuildMemberInstanceExpression(memberInfo));
+                fieldType.Arguments = null;
+                fieldType.Resolver = resolver;
+                fieldType.StreamResolver = null;
+            }
+            else if (memberInfo == null)
+            {
+                throw new ArgumentNullException(nameof(memberInfo));
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException(nameof(memberInfo), "Member must be a field, property or method.");
+            }
+        }
+
+        /// <summary>
+        /// Determines if the type is an <see cref="IObservable{T}"/> or task that returns an <see cref="IObservable{T}"/>.
+        /// </summary>
+        private static bool IsObservable(Type type)
+        {
+            if (!type.IsGenericType)
+                return false;
+
+            var g = type.GetGenericTypeDefinition();
+            if (g == typeof(IObservable<>))
+                return true;
+            if (g == typeof(Task<>) || g == typeof(ValueTask<>))
+                return IsObservable(type.GetGenericArguments()[0]);
+            return false;
+        }
+
+        /// <summary>
+        /// Returns a list of properties, methods or fields that should have fields created for them.
+        /// <br/><br/>
+        /// Unless overridden, returns a list of public instance readable properties and public instance methods
+        /// that do not return <see langword="void"/> or <see cref="Task"/>
+        /// including properties and methods declared on inherited classes.
+        /// </summary>
+        internal static IEnumerable<MemberInfo> GetRegisteredMembersForOutput<TSourceType>(Expression<Func<TSourceType, object?>>[]? excludedProperties)
+        {
+            if (typeof(TSourceType).IsInterface)
+            {
+                var types = typeof(TSourceType).GetInterfaces().Append(typeof(TSourceType));
```

Comment:
Multiple interfaces may declare members with the same name and the same or different return type. The is no such problem for `typeof(TSourceType).GetProperties(flags)`.

Reply:
That is true; it would cause an error when adding a second field with the same name to the interface.  I could write a test to verify this, but I did not see it worth the trouble to do so.  This of course assumes that the user does not tag the second field with the `[Name]` attribute to change the name within the generated graph type.

Note that if there is a complex inheritance structure, where multiple paths lead to the same member, this does not matter because `GetInterfaces()` return all inherited interfaces, and members cannot be redefined on children without the `new` keyword, which would of course break the graph because of a second field with the same name.  I added `GraphTypeCanImplementMultipleInterfaces` to demonstrate this, and there's already a multiple tests demonstrating inherited interfaces' members are used (as `IAnimal` inherits from `IObject`).

So unless you would like additional tests, I do not see any changes necessary to this code.

Reply:
It is enough for me that you are aware of the problem.